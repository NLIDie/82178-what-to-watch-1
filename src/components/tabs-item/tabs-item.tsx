import * as React from "react";

// Types
import {TProps} from "./tabs-item.types";

export function TabsItem(props: TProps): JSX.Element {
  const {onClick, tab, isActive} = props;

  const handleTabClick = (event: React.SyntheticEvent<HTMLElement>) => {
    event.preventDefault();

    onClick(tab, event);
  };

  return (
    <li
      className={`movie-nav__item ${isActive ? `movie-nav__item--active` : ``}`}
      onClick={handleTabClick}
    >
      <a href="#" className="movie-nav__link">
        {tab}
      </a>
    </li>
  );
}
