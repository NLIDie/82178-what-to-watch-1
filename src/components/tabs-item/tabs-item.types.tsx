import * as React from 'react';

export type TProps = {
  tab: string;
  isActive: boolean;
  onClick: (tab: string, event: React.SyntheticEvent<HTMLElement>) => void,
}
