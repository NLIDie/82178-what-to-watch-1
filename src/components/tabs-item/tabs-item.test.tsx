import * as React from "react";
import {create} from "react-test-renderer";

// Components
import {TabsItem} from "./tabs-item";

describe(`Components/TabItem`, () => {
  it(`Correct renderer the component`, () => {
    const tree = create(<TabsItem tab="tab1" onClick={jest.fn()} isActive />);

    expect(tree).toMatchSnapshot();
  });
});
