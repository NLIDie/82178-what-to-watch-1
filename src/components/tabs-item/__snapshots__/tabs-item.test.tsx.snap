// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/TabItem Correct renderer the component 1`] = `
<li
  className="movie-nav__item movie-nav__item--active"
  onClick={[Function]}
>
  <a
    className="movie-nav__link"
    href="#"
  >
    tab1
  </a>
</li>
`;
